### PR TITLE
mew

### PR DESCRIPTION
**Pull request by mostafa-kamel11**
new pull mew

### CODE REVIEW

File: gridkey.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import re 
+def get_meta_from_contract(contract_cont):
+    contract_note_no = re.findall("Contract Note No.*?:(.*)",contract_cont)[0]
+    trade_date = re.findall ("TRADE DATE\s*?:\s*(\d{2}-\d{2}-\d{4}).*",contract_cont)[0]
+    settlemnt_no = re.findall ("SETTLEMENT NO.\s(\d*).*",contract_cont)[0]
```
Comment (OmarElsebaey001) [inline gridkey.py:5]: this has to be this way xzxxxx

File: gridkey.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        for record in transaction_table :
+            records_list = record.split()
+            name = " ".join(records_list[5:len(records_list)-6])
+            del records_list[5:len(records_list)-6]
+            records_list.insert(5,name)
+            transactions.append(records_list)
+    return transactions
+def extract_meta_from_table (seperated_table):
+    meta_detailed_table = []
+    for table in seperated_table:
```
Comment (OmarElsebaey001) [inline gridkey.py:29]: another review

File: gridkey.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    meta_summary = extract_meta_summary_table(summary_table)
+    transaction_summary = exract_transactin_summary_table(summary_table)
+    return (transaction_summary,meta_summary)
+
+def get_data_from_contract (contract_path):
+    contract_handler = open (contract_path,"r")
+    contract_content = contract_handler.read()
+    meta_contract = get_meta_from_contract(contract_content)
+    seperate_detailed_table = seperate_table_from_contract(contract_content)
+    detailed_table = get_detailed_table(seperate_detailed_table)
```
Comment (OmarElsebaey001) [inline gridkey.py:63]: few changes

**Review by OmarElsebaey001 — CHANGES_REQUESTED** [review]
I have done the following suggestions

File: gridkey.py (hunk 3 above)
Comment (mostafa-kamel11) [reply gridkey.py:63]: ok

**Pull request closed without merging**